lib/cloud/imds/azure/imds_test.go

```diff
@@ -205,13 +206,20 @@ func TestGetInstanceInfo(t *testing.T) {
 		},
 		{
 			name:         "request error",
-			stausCode:    http.StatusNotFound,
+			statusCode:   http.StatusNotFound,
+			errAssertion: require.Error,
+		},
+		{
+			name:         "empty body returns an error",
+			statusCode:   http.StatusOK,
 			errAssertion: require.Error,
 		},
 	} {
+		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-				w.WriteHeader(http.StatusOK)
+				w.WriteHeader(tc.statusCode)
```

Comment:
Let's assert the returned error below (`tc.errAssertion(t, err)`) beyond just Error/NoError. I'm not confident that we are hitting the correct errors, given that this wasn't returning the expected status code before.

I like require.ErrorContains (asserting a part of the error string) for disambiguation. You can pair it with a "trace.IsFoo" check if you want to be extra sure about it.